### ISSUE TITLE
Organize multiple notes in note view

### ISSUE THREAD

**Issue by InnayTool** [hedgedoc/hedgedoc#1001]
**Idea:**
Have a method to organize notes as groups or folders in the note view (not in history or explore page #994). There could be a sidebar or something.

**Use-cases:**
- create something like a book Issue ([Book Mode](#300)
    - "A sort of book/multi-page-menu driven output to facilitate building documentation"
 https://github.com/hedgedoc/hedgedoc/issues/429 
- menu for multiple notes

**Solutions:**
- may be implemented with folders, so that all notes in a folder are displayed in one sidebar or something

**Comment by DerMolly** [comment]
I think coupling folders and book mode together could be problematic, as people may want to put notes into folders without having a permanent sidebar.
A collapseable sidebar could be the solution here.

**Comment by DerMolly** [comment]
Also why should there 

> Have a method to organize notes as groups or folders in the note view (not in history or explore page #994). There could be a sidebar or something.

I think it's only logical to show notes organized in folders also in folders in the history. The two problems that a rise from that are:

- the history is not the same for everybody, so we would need to find a way to put folder information into the history api.
  -  either folder information are put into each note and therefore into each history entry and the frontend needs to assemble a list of folders to show on each history call
  - or there needs to be a separate folder api call where the frontend get's all folders the user should be shown in the frontend. Since we most likely need such a folder list to put notes into folders. I prefer this.
- If a user views a note that's in a folder they may not want to be shown the sidebar.
  - Solution: make the sidebar collapsible.

**Comment by InnayTool** [comment]
No you misunderstood. I have tried to make the statement clearer. This issue should not handle how the history or explore page or something shows notes and folders.

**Comment by DerMolly** [comment]
Well that (at least in my opinion) is inherently linked and should not be considered separately.

**Comment by InnayTool** [comment]
There is issue #994 which handles this. I think it's to much to merge them together and they are separate things to implement. I updated the description again to make clear that implications to this page are not banned from this issue.

**Comment by DerMolly** [comment]
> they are separate things to implement

#### From a backend standpoint?

No and if it was, the implementation would be greatly flawed.

#### From a frontend standpoint?

yes, but that would not be tracked here anyway.

**Comment by InnayTool** [comment]
Yes it's probably more a frontend view, but were to track overall features of the project?
I think we decided before to have discussions on features in this repo.

**Comment by davidmehren** [comment]
The first post has been updated with a new proposal called "notebook mode".

**Comment by davidmehren** [comment]
To elaborate a bit: In contrast to the [folder proposal](https://github.com/hedgedoc/hedgedoc/issues/1002#issuecomment-792049138), a notebook has a common structure for everyone that has access to it and is built for collaboration with permission management and share features. Folders on the other hand are intended for personal organization.

Not allowing a note to appear in multiple notebooks follows a real notebook: Either you put a page in notebook A or in notebook B, if you want to have it appear in both, you need to copy it. We hope this makes it easier to understand this feature for new users. Still, adding some kind of 'wormhole' which allows a note to appear in multiple places is still possible at a later time.

**Comment by mike-hales** [comment]
My project has a need for book-scale documentation - a members' handbook for a multistakeholder coop, [meet.coop](https://www.meet.coop/), which operates a platform that runs BigBlueButton video-conferencing, and is committed to open source. From the above, it's definitely the <notebook> rather than <folder> approach that approximates our need.

At the same time, I get the <folder> requirement too. My personal commitment here - in terms of personally organising a shared universe of material (along the lines of a 'commonplace book') - is to work in **federated wiki**. Developed subsequently by Ward Cunningham, who also invented 'orthodox' wiki. Fedwiki has a 'chorus of voice' orientation, as distinct from the consensus orientation of orthodox wiki (and perhaps, of collaborative writing in Hedgedoc?). Federation, rather than convergence into 'single voice'. The two complement each other?

**Comment by almereyda** [comment]
OT: @mike-hales Meanwhile, before the HedgeDoc community allows to mirror the HackMD book mode, you could have a look at https://github.com/outline/outline/ (pluggable authentication needs to be self-built, based on Passport.js), https://nextcloud.com/blog/nextcloud-22-introduces-knowledge-management/ or https://github.com/documize/community. Additionally, best in class documentation is also provided by https://github.com/documize/community. If you need this with real-time collaboration enabled, probably CryptPad is your way to go, since they implement folders as described in #1002.

---

OP: A Notebook could be a special view on a kind of document that merely contains an outline of (relative) links to certain pads, which can be rendered as a documentation website similar to what Hackfoldr did back in the day, or a special view on a hierarchy of (relative) links to certain pads that is encoded in the YAML frontmatter.

**Comment by ThetaEdra** [comment]
Currently missing Folder/Notebook structures is the only thing keeping me from using hedgedoc. I used CodiMD before but was disappointed that the folder structure was not available in the open source version. Jooplin and Trillium seem like great tools for organizing my notes and ideas but they all miss an imoportant point. I might want to share notes publicly and i might want someone else to collaborate on a file with me. All those tools either lack the structure and possibilities of offline editors or dont offer online sharing for single notes. I would appreciate this feature and i wonder why there are so many similiar note taking apps out there that are all similiar and still missing the point. 
I also strongly dislike the landing page and the way users are handled.

Proper Permission System similiar to the screenshots of Plainpad where you can manage users and their permissions, live/co-editing from hedgedoc, E2E encrypted notes that can be shared and live edited via URL and get Clientside decrypted and structur from Trillium or Foam would truly be the best editor out there. Seems like i gotta keep searching for now.

**Comment by theo-gaillard** [comment]
Hello, I would like to know if there is an update concerning this issue or a feature that is similar to book mode coming soon ?

**Comment by mrdrogdrog** [comment]
@theo-gaillard 
Not in the near future. This issue is marked as "Version 2.x" . This means that we will work on it after the 2.0 release.
